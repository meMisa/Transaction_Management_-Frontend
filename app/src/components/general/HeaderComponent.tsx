// node modules
import * as React from 'react';
import { Layout, Avatar, Typography, Popover } from 'antd';
import { UserOutlined } from '@ant-design/icons';
// images
import profile_icon from 'assets/images/profile_icon.png';
// styles
import 'styles/header.css';

const { Header } = Layout;
const { Text, Title } = Typography;

const HeaderComponent = () => {
  const contetnt = (
    <div className="text-align-center padding-1">
      <Avatar size={64} icon={<UserOutlined />} src={profile_icon} />
      <Title level={4} className="margin-top-1">
        Misa Qarahqani
      </Title>
      <Text className="display-block">misa64gh@gmail.com</Text>
    </div>
  );

  return (
    <Header className="header direction-rtl">
      <Popover content={contetnt} trigger="click" placement="topRight">
        <Avatar
          size={'large'}
          // Popover
          icon={<UserOutlined />}
          src={profile_icon}
          className="cursor-pointer"
        />
      </Popover>
    </Header>
  );
};
export default HeaderComponent;
