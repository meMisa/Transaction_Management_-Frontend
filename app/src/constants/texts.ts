export const TRANSACTIONS = 'Transactions';
export const SETTINGS = 'Settings';
export const ITEMS = 'Items';
export const NEW_TRANSACTION = 'New Transaction';
export const ADD_NEW_TRANSACTION = 'Add New Transaction';
export const ACCOUNT_ID = 'Account Id';
export const AMOUNT = 'Amount';
export const THIS_FIELD_IS_REQUIRED = 'this field is required';
export const SUCCESS = 'Success';
export const ERROR = 'Error';
export const GOOD_LOCK = 'GOOD_LOCK';
export const GET_THE_TRANSACTIONS_LIST_SUCCESSFULLY =
  'get the transaction list successfully.';
export const SOMETHING_IS_WRONG = 'Ops! something is wrong!please try later.';
export const EDIT_NEW_TRANSACTION = 'View the Transaction';
export const NEW_TRANSACTION_ADDED_SUCCESSFULLY =
  'new record added successfully.';
export const BALANCE = 'Balance:';
