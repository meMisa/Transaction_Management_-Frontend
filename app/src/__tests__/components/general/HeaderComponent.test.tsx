import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import HeaderComponent from 'components/general/HeaderComponent';
import { BALANCE } from 'constants/texts';

const userInfo = {
  account_id: 'a31a9c62-001b-40c5-8903-48318555b4e7',
  balance: 100,
};

afterEach(() => {
  cleanup();
});

describe('Test', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => {
        return {
          matches: true,
          addListener: jest.fn(),
          removeListener: jest.fn(),
        };
      }),
    });
  });
  test('should render HeaderComponent component correctly', () => {
    const Header = render(<HeaderComponent {...userInfo} />);
    const headerComponent = screen.getByTestId('header');
    expect(headerComponent).toBeInTheDocument();
    expect(headerComponent).toHaveTextContent('');
    expect(headerComponent).toContainHTML('<img src="profile_icon.png" />');
  });

  it('Text in state is changed when button clicked', () => {
    const { getByText } = render(<HeaderComponent {...userInfo} />);

    // expect(getByText(BALANCE).textContent).toBe(BALANCE)

    // fireEvent.click(getByText("State Change Button"))
    //
    // expect(getByText(/Initial/i).textContent).toBe("Initial State Changed")
  });

  test('match snapshot', () => {
    const tree = renderer.create(<HeaderComponent {...userInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
