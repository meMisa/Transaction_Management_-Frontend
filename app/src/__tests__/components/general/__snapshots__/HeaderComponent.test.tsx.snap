// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test match snapshot 1`] = `
<header
  className="ant-layout-header header direction-rtl"
  data-testid="header"
>
  <span
    className="ant-avatar ant-avatar-lg ant-avatar-circle ant-avatar-image ant-avatar-icon cursor-pointer"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
  >
    <img
      onError={[Function]}
      src="profile_icon.png"
    />
  </span>
</header>
`;
